components/interop/interop-service/src/main/kotlin/net/corda/interop/InteropProcessor.kt

```
@@ -35,12 +35,12 @@ class InteropProcessor(cordaAvroSerializationFactory: CordaAvroSerializationFact
         events: List<Record<String, AppMessage>>
     ): List<Record<*, *>> {
         val outputEvents = mutableListOf<Record<*, *>>()
-        events.forEach { appMessage ->
-            val unAuthMessage = appMessage.value?.message
+        events.forEach { record ->
+            val unAuthMessage = record.value?.message
             //TODO temporary using UnauthenticatedMessage instead of AuthenticatedMessage
             if (unAuthMessage != null && unAuthMessage is UnauthenticatedMessage && unAuthMessage.header.subsystem == SUBSYSTEM) {
                 val header = with(unAuthMessage.header) { CommonHeader(source, destination, null, messageId) }
-                getOutputRecord(header, unAuthMessage.payload, appMessage.key)?.let { outputRecord ->
+                getOutputRecord(header, unAuthMessage.payload, record.key)?.let { outputRecord ->
```

`mapNotNull` is good for when you're transforming one list into another, and want to filter and map all in one go:

```kotlin
    override fun onNext(
        events: List<Record<String, AppMessage>>
    ): List<Record<*, *>> = events.mapNotNull { (key, value) ->
        val unAuthMessage = value?.message

        //TODO temporary using UnauthenticatedMessage instead of AuthenticatedMessage
        if (unAuthMessage == null ||
             unAuthMessage !is UnauthenticatedMessage ||
             unAuthMessage.header.subsystem != SUBSYSTEM) return@mapNotNull null
         
        val header = with(unAuthMessage.header) { CommonHeader(source, destination, null, messageId) }
        getOutputRecord(header, unAuthMessage.payload, key)
    }
```

When/why might `getOutputRecord` return null?

---

Partially is desired, as this component is not the only one reading from Kafka queue. We specifically read `SUBSYSTEM=="Interop"`. I see, yes, we could silently return null if it's not our subsystem, however warn if it's our subsystem but we have some of wrong data e.g. payload is null.

---

Code change as above to stream like processing, added TODO about for potentially more considerate warning if we would get `SUBSYSTEM=Interop` but inside an unexpected message type, however it's tricky to check this, as different message types do not share the same interface, but anyway there is 3-5 different potential types.